engine/script/src/script_vmath.cpp

```diff
@@ -300,8 +300,9 @@ namespace dmScript
         size_t size = 0;
         const char* s = luaL_checklstring(L, 1, &size);
         Vectormath::Aos::Vector3* v = CheckVector3(L, 2);
-        char* buffer = new char[size + 61];
-        dmSnPrintf(buffer, size, "%svmath.%s(%f, %f, %f)", s, SCRIPT_TYPE_NAME_VECTOR3, v->getX(), v->getY(), v->getZ());
+        char* buffer = new char[size + 68];
+        // Use same format as Lua when converting number to string (from LUA_NUMBER_FMT in luaconf.h)
+        dmSnPrintf(buffer, size + 68, "%svmath.%s(%.14g, %.14g, %.14g)", s, SCRIPT_TYPE_NAME_VECTOR3, v->getX(), v->getY(), v->getZ());
```

Comment:
Fyi, 32 bit floating point numbers doesn't have 14 bits of precision.

Also, what is 68 based off of? It increased by 7? 7 is not a multiple of 3, so I'm curious.

Like Lua, I'd recommend constants (e.g. at the top of the file), where we can also document them

Reply:
> 32 bit floating point numbers doesn't have 14 bits of precision

The %.14g format comes from here:

https://github.com/defold/defold/blob/dev/engine/lua/src/dmsdk/lua/luaconf.h#L522.

We use lua_pushfstring() for Vector3_tostring() so I thought it best to use the same format specifier for our own conversion.

NOTE: We call `lua_pushfstring` with "%f" which is Lua's own format specifier which under the hood translates to "%.14g".

> Also, what is 68 based off of?

Worst case scenario when converted to String is (if I'm not mistaken) something like this:

`vmath.vector3(-10.010000228882, -10.010000228882, -10.010000228882)`

That is 67 characters, and then we need another to have room for the terminating null character.

Reply:
You specify 14 decimals, but in your example there is only 12?

My main point is that there should be a constant defining how many characters a float can take.
E..g "MAX_CHARS_PER_FLOAT = x"
and
MAX_VEXTOR3_STRING = MAX_CHARS_PER_FLOAT * 3 + ...

Reply:
Hmm, you're right.